test/client/contact/ContactUtilsTest.ts

```diff
@@ -5,149 +5,180 @@ import {createContactMailAddress} from "../../../src/api/entities/tutanota/Conta
+		compare("", "", null, "", "", "Bob", true, 1)
+		compare("", "", "Bob", "", "", "Alf", true, 1)
+		// first and last name
+		compare("", "Alf", null, "Alf", "Bob", null, true, 1)
+		compare("Alf", "Bob", null, "Bob", "Alf", null, true, -1)
+		compare("Alf", "Bob", null, "", "Bob", null, true, -1)
+		compare("Alf", "", null, "Alf", "Bob", null, true, 1)
+		// mixed
+		compare("", "Bob", null, "", "", "Alf", true, -1)
+		compare("Bob", "", null, "", "", "Alf", true, -1)
+		compare("Alf", "Bob", "Bob", "Alf", "Bob", "Alf", true, 1)
+		compare("Alf", "Bob", null, "", "", "Alf", true, -1)
+		// none or same
+		compare("", "", null, "", "", null, true, 1) // reverse id
+
+		compare("Alf", "Bob", "Bob", "Alf", "Bob", "Bob", true, 1) // reverse id
+
+		compare("ma", "p", "aa", "Gump", "Forrest", "aa", true, 1) // reverse id
+	})
+	o("compareContacts by last name", () => {
+		// only first name
+		compare("Alf", "", null, "", "", null, false, -1)
+		compare("Alf", "", null, "Bob", "", null, false, -1)
+		compare("", "", null, "Bob", "", null, false, 1)
+		compare("Bob", "", null, "Alf", "", null, false, 1)
+		// only last name
+		compare("", "Alf", null, "", "", null, false, -1)
+		compare("", "Alf", null, "", "Bob", null, false, -1)
+		compare("", "", null, "", "Bob", null, false, 1)
+		compare("", "Bob", null, "", "Alf", null, false, 1)
+		// only mail address
+		compare("", "", "Alf", "", "", null, false, -1)
+		compare("", "", "Alf", "", "", "Bob", false, -1)
+		compare("", "", null, "", "", "Bob", false, 1)
+		compare("", "", "Bob", "", "", "Alf", false, 1)
+		// first and last name
+		compare("", "Alf", null, "Alf", "Bob", null, false, -1)
+		compare("Alf", "Bob", null, "Bob", "Alf", null, false, 1)
+		compare("Alf", "Bob", null, "", "Bob", null, false, -1)
+		compare("Alf", "", null, "Alf", "Bob", null, false, 1)
+		// mixed
+		compare("", "Bob", null, "", "", "Alf", false, -1)
+		compare("Bob", "", null, "", "", "Alf", false, -1)
+		compare("Alf", "Bob", "Bob", "Alf", "Bob", "Alf", false, 1)
+		compare("Alf", "Bob", null, "", "", "Alf", false, -1)
+		// none or same
+		compare("", "", null, "", "", null, false, 1) // reverse id
+
+		compare("Alf", "Bob", "Bob", "Alf", "Bob", "Bob", false, 1) // reverse id
+
+		compare("ma", "p", "aa", "Gump", "Forrest", "aa", false, 1) // reverse id
+	})
+	o("formatNewBirthdayTest", function () {
+		lang.setLanguage({
+			code: "en",
+			languageTag: "en",
+		})
+		lang.updateFormats({})
+		let bday = createBirthday()
+		bday.day = "12"
+		bday.month = "10"
+		bday.year = "2009"
+		o(formatBirthdayNumeric(bday)).equals("10/12/2009")
+		bday.day = "9"
+		bday.month = "07"
+		bday.year = null
+		o(formatBirthdayNumeric(bday)).equals("7/9")
+		bday.day = "09"
+		bday.month = "7"
+		bday.year = null
+		o(formatBirthdayNumeric(bday)).equals("7/9")
+		bday = createBirthday()
+		bday.day = "12"
+		bday.month = "10"
+		bday.year = "2009"
+	})
+
+	o("formatBirthdayNumeric", () => {
```

Comment:
We generally write case definition using `function()` instead of arrow function because ospec is using it for some state tracking (yeah...) so if you could replace this one and other in this file?